tests/conftest.py

```diff
@@ -5975,3 +5975,9 @@ def taxi_test_file_directory():
         __file__,
         os.path.join("test_sets", "taxi_yellow_tripdata_samples", "first_3_files/"),
     )
+
+
+@pytest.fixture()
+def test_df_pandas():
+    test_df: pd.DataFrame = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    return test_df
```

Comment:
This is shared by both `DataContext`-level and `DataConnector`-level tests, and has been moved back to our base `conftest.py`.

Reply:
@Shinnnyshinshin Question: Do `batch_identifiers` at the `DataConnector` level really need to get deprecated?..

Reply:
This was the request from @jcampbell from our last discussion. 

That said, I do realize it hasn't gone through a formal review process, so would there be world where we could merge this PR with the deprecation warning commented-out (so that Nimbus can use the functionality), to be enabled once we confirm whether it should stay?